docs/release-notes/mongoid-7.4.txt

```diff
@@ -199,3 +240,130 @@ an index where it wouldn't before:
 .. code-block:: ruby
 
   Band.desc(:name).count
+
+
+Return ``0`` When Aggregating Empty Result Sets
+-----------------------------------------------
+
+Mongoid 7.4 with the ``Mongoid.broken_aggregables`` option set to ``false``
+will return ``0`` from the ``sum`` method over an empty result set, for example:
+
+.. code-block:: ruby
+
+  Product.where(impossible_condition: true).sum(:price)
+  # => 0
+
+Mongoid 7.3 and Mongoid 7.4 with the ``Mongoid.broken_aggregables`` option
+set to ``true`` (the default) returns ``nil`` in this case.
+
+.. note::
+
+  The value of ``Mongoid.broken_aggregables`` option will change to ``false``
+  by default in Mongoid 8.0.
+
+
+Correct Update Behavior When Replacing Association
+--------------------------------------------------
+
+Mongoid 7.4 with the ``Mongoid.broken_updates`` option set to ``false``
+will correctly persist an ``embeds_one`` association target that is set to nil
+and then to a non-nil value, for example:
+
+.. code-block:: ruby
+
+  class Canvas
+    include Mongoid::Document
+    
+    embeds_one :palette
+  end
+  
+  canvas.palette = palette
+  canvas.palette = nil
+  canvas.palette = palette
+
+In Mongoid 7.3 and earlier, and in 7.4 with the ``Mongoid.broken_aggregables``
+option set to ``true`` (the default), ``canvas.palette`` would be ``nil`` when
+we would expect it to be ``palette``.
+
+.. note::
+
+  The value of ``Mongoid.broken_update`` option will change to ``false``
+  by default in Mongoid 8.0.
+
+
+Correct Logical ``and`` Query Generation
+----------------------------------------
+
+Mongoid 7.4 with the ``Mongoid.broken_and`` option set to ``false``
+will preserve existing conditions when using ``and`` to add new conditions
+to a query when the same operator is used on the same field multiple times.
+For example, in the following query:
+
+.. code-block:: ruby
+
+  Band.where(id: 1).and({year: {'$in' => [2020]}}, {year: {'$in' => [2021]}}).where(id: 2)
+
+Mongoid 7.4 with the ``Mongoid.broken_and`` option set to ``false`` will
+generate the following criteria:
+
+.. code-block:: ruby
+
+  #<Mongoid::Criteria
+    selector: {"_id"=>1, "year"=>{"$in"=>[2020]}, "$and"=>[{"year"=>{"$in"=>[2021]}}, {"_id"=>2}]}
+    options:  {}
+    class:    Band
+    embedded: false>
+
+In Mongoid 7.3 and earlier, and in 7.4 with the ``Mongoid.broken_and``
+option set to ``true`` (the default), the following criteria would be
+generated instead which omit the {"$in" => [2021]} condition:
```

Comment:
"omits"?

It's weird because, yes, criteria is a plural word, but for all intents and purposes we treat it like singular.... Doesn't actually matter just whatever sounds better

Reply:
Actually I think you're right on reading it again. Resolving.